projects/cm_mcu/I2CCommunication.c

```diff
@@ -155,19 +154,19 @@ int apollo_i2c_ctl_reg_w(uint8_t device, uint8_t address, uint8_t nbytes_addr, u
   configASSERT(p_sMaster != NULL);
 
   // first byte (if write to one of five clcok chips) or two bytes (if write to EEPROM) is the register, others are the data
-  uint8_t data[MAX_BYTES_ADDR + MAX_BYTES];
-  for (int i = 0; i < MAX_BYTES_ADDR; ++i){
-	data[i] = (packed_reg_address >> (nbytes_addr - 1 - i) * 8) & 0xFFUL;
+  uint8_t data[nbytes_addr + MAX_BYTES];
+  for (int i = 0; i < nbytes_addr; ++i){
+	data[i] = (packed_reg_address >> (nbytes_addr - 1 - i) * 8) & 0xFF;
     if (data[i] != 0) ++nbytes; // to account for the register address
   }
   // pack the bytes into the data array, offset by
   // one or two due to the address
-  for (int i = MAX_BYTES_ADDR; i < MAX_BYTES + MAX_BYTES_ADDR; ++i) {
-    data[i] = (packed_data >> (i) * 8) & 0xFFUL;
+  for (int i = nbytes_addr; i < MAX_BYTES + nbytes_addr; ++i) {
+    data[i] = (packed_data >> (i) * 8) & 0xFF;
```

Comment:
Looking at this I think it does not work. for `nbytes_addr=2` the first data byte is downshifted 2 bytes but I think you want it shifted by 0 bytes, right? I'm not sure if this was correct in original version or not...

Reply:
Yes, you're right. This should be:
`data[i] = (packed_data >> (i-nbytes_addr)*8) & 0xFF` just as what the original did https://github.com/apollo-lhc/cm_mcu/blob/master/projects/cm_mcu/I2CCommunication.c#L158-L159.